### PR TITLE
Formula installer bug fix

### PR DESCRIPTION
**Pull request by xu-cheng**
This should help to solve https://github.com/Homebrew/homebrew/pull/44231#issuecomment-142518526

### CODE REVIEW

File: Library/Homebrew/formula_installer.rb [hunk 1]
```diff
@@ -97,7 +97,7 @@ def install_bottle_for?(dep, build)
   end
 
   def prelude
-    verify_deps_exist unless ignore_deps?
+    verify_deps_exist unless ignore_deps? || is_a?(DependencyInstaller)
```
Comment (mikemcquaid) [inline Library/Homebrew/formula_installer.rb]: Mind making this a `if`? I always find `unless ... or` hard to parse.

Comment (mikemcquaid) [inline Library/Homebrew/formula_installer.rb]: And same below. I wonder if you want a dedicated method for them all?

**Comment by mikemcquaid** [comment]
Nice work :+1:

File: Library/Homebrew/formula_installer.rb (hunk 1 above)
Comment (xu-cheng) [inline Library/Homebrew/formula_installer.rb]: What method name should we call it? How about `skip_deps_check?`? So we will have:
```ruby
class FormulaInstaller
  def skip_deps_check?
    ignore_deps?
  end
end

class DependencyInstaller < FormulaInstaller
  def skip_deps_check?
    true
  end
end
```

Comment (mikemcquaid) [inline Library/Homebrew/formula_installer.rb]: Yeh, something like that looks great :+1:

**Comment by xu-cheng** [comment]
Updated, also fixed a new bug: prefix is not cleanup when pouring bottle failed.

**Comment by mikemcquaid** [comment]
Looking good :+1:

**Comment by xu-cheng** [comment]
@mikemcquaid Travis seems to be confused on which commits to test. Instead of testing this PR, it's testing the formula updated in current time period  https://travis-ci.org/Homebrew/homebrew/jobs/82146619

**Comment by mikemcquaid** [comment]
@xu-cheng you can look at the:
```
name: Homebrew/homebrew/pull/44325-ff232e29f2758c887b56dc331999e6b737ad8565
url: https://github.com/Homebrew/homebrew/pull/44325
hash: 
diff_start_sha1: 628bde1ca9ad50715b7e394ac0381e7e40e22a35
diff_end_sha1: ff232e29f2758c887b56dc331999e6b737ad8565
```

debugging information which should help debug that.

**Comment by xu-cheng** [comment]
Will looking into it tomorrow.
